src/audio/SDL_audio.c

```diff
@@ -1239,7 +1239,7 @@ SDL_bool SDL_RecordingAudioThreadIterate(SDL_AudioDevice *device)
                 void *output_buffer = device->work_buffer;
 
                 // I don't know why someone would want a postmix on a recording device, but we offer it for API consistency.
-                if (logdev->postmix) {
+                if (logdev->postmix || (logdev->gain != 1.0f)) {
```

Comment:
Do we want an epsilon check here?

Reply:
1.0f can be represented exactly in floating point, so the literal "don't adjust the volume" value shouldn't do the processing, but probably anything else should...?